uqcsbot/member_counter.py

```diff
@@ -37,16 +38,22 @@ async def on_ready(self):
             self.member_count_channel = None # TODO maybe end this cog or something similar
             return
 
+        bot_member = self.bot.uqcs_server.get_member(self.bot.user.id)
+        permissions = self.member_count_channel.permissions_for(bot_member)
+        if not permissions.manage_channels:
+            logging.warning(
+                f"Bot does not have the permission to manage the #Member Count channel. The current permissions are {permissions}. The bot user is {bot_member}.")
         await self.attempt_update_member_count_channel_name()
 
     @app_commands.command(name="membercount")
     async def member_count(self, interaction: discord.Interaction):
         """ Display the number of members """
-        new_members = filter(
-            lambda member: member.joined_at > datetime.now() - self.NEW_MEMBER_TIME,
-            interaction.guild.members
-        )
-        await interaction.response.send_message(f"There are currently {interaction.guild.member_count} members in the UQCS discord server, with {len([new_members])} joining in the last 7 days.")
+        new_members = [
+            member
+            for member in interaction.guild.members
+            if member.joined_at > datetime.now(tz=ZoneInfo("Australia/Brisbane")) - self.NEW_MEMBER_TIME
+        ]
+        await interaction.response.send_message(f"There are currently {interaction.guild.member_count} members in the UQ Computing Society discord, with {len(new_members)} joining in the last 7 
```

Comment:
This is broken...

Reply:
Oops. Merging with main got rid of the end of that that line by mistake. Fixed by appending ` days")`. Should be fixed in latest commit.